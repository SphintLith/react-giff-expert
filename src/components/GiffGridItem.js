import React from 'react'
import PropTypes from 'prop-types'

const GiffGridItem = ( {title,url} ) => {
  return (
    <div className="card animate__animated animate__fadeInDown">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}
GiffGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
export default GiffGridItem