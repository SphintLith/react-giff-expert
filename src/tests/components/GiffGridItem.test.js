import { shallow } from 'enzyme'
import GiffGridItem from '../../components/GiffGridItem'

describe('Pruebas en <GiffGridItem />', () => {
  const title = 'A Title'
  const url = 'http://localhost/algo.jpg'
  let wrapper = shallow(<GiffGridItem title={ title } url={ url } />)
  beforeEach(() => {
    wrapper = shallow(<GiffGridItem title={ title } url={ url } />)
  })

  test('Debe mostrar <GiffGridItem /> correctamente', async () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('debe de tener un parrafo con el titulo', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)
  })
  test('debe de tener una imagen con el alt y url de los props', () => {
    const img = wrapper.find('img')
    //console.log(img.props())
    expect(img.props().src).toBe(url)
    expect(img.props().alt).toBe(title)
  })
  test('Debe tener animate__fadeIn', () => {
    const div = wrapper.find('div')
    const className = div.props().className
    expect(className.includes('animate__fadeInDown')).toBe(true)
  })
})