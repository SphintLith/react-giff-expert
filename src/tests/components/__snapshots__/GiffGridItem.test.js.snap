// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiffGridItem /> Debe mostrar <GiffGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="A Title"
    src="http://localhost/algo.jpg"
  />
  <p>
    A Title
  </p>
</div>
`;
